protostar/cheatable_starknet/cheaters/expect_events_controller.py

```diff
@@ -0,0 +1,166 @@
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Optional
+
+from protostar.starknet import Address, CairoData, Selector, ReportedException
+
+if TYPE_CHECKING:
+    from protostar.testing import Hook
+    from protostar.cheatable_starknet.cheatables.cheatable_starknet_facade import (
+        CheatableStarknetFacade,
+    )
+    from protostar.cairo_testing import CairoTestExecutionState
+
+
+@dataclass
+class Event:
+    from_address: Address
+    key: Selector
+    data: Optional[CairoData] = None
+
+
+@dataclass
+class EventMatching:
```

Comment:
This should be Union type instead

Reply:
Why? With this approach, I don't have to remember to "register" or "unregister" concrete `EventMatching`. This is also consistent with `test_results.py`.

Reply:
For me it seems more natural to represent composite types as union instead of subtyping, this should be marked as [[Nitpick]]